src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -295,7 +295,7 @@ private SimpleMonitor EnsureMonitorInitialized()
         private void OnSerializing(StreamingContext context)
         {
             EnsureMonitorInitialized();
-            _monitor._busyCount = _blockReentrancyCount;
+            _monitor!._busyCount = _blockReentrancyCount;
```

Comment:
Nit but just as an FYI as I just learnt it as well. In `EnsureMonitorInitialized` we could change it to: 

```cs
        private SimpleMonitor EnsureMonitorInitialized() => _monitor ??= new SimpleMonitor(this);